### ISSUE TITLE
RPM fluctuation at constant throttle when field weakening is activated... Get worse as battery voltage drops.

### ISSUE THREAD

**Issue by papagino007** [electricrt/nxt_firmware_updates#10]
I am currently running firmware 6.4.5E and noticed that I get a lot of RPM fluctuation at constant throttle position once Field Weakening is activate. This problem is also observed by lots of other people. This is barely noticeable with a fully charged 72v battery (84V) but this problem worsen when battery voltage get lower (somewhere below 74V) and keeping constant speed with a fixed throttle position is almost impossible when battery voltage drops below 70V.

I have seen somewhere that the fluctuation is caused by an improperly tuned current regulator. Apparently, raising the current regulator Ki setting seems to smooth out the fluctuations.

(I have a cruise control module on my bike but can't use it for this reason)

Would it be possible to fix this issue with the new RC6.50 Beta firmware?

Let me know if you need a beta tester....

This is Dan by the way.

**Comment by electricrt** [comment]
Ok let's address one at a time. 

Field Weakening issue:

When does the rpm begin fluctuating?

Would you describe it such as the motor is oscillating at a certain rpm even if you maintain the same throttle position?

**Comment by papagino007** [comment]
It is the RPM that's going up and down by its itself with constant throttle position, this has nothing to do with speed limiting as it will happen at any speed regardless of gear selection. No speed limit set for Sport mode... I mostly use Torque mode, but I think it is also doing this in Speed mode.

**Comment by electricrt** [comment]
Gotcha, this is only when field weakening is active correct? Or does it happen at literally any rpm?

**Comment by papagino007** [comment]
If you turn down FW, say less than 15 or 20 in the settings, it is barely noticeable. Turning it OFF will stop the fluctuation completely. It is when you set it at 38 - 40 that it become really bad. The lower the battery voltages is, the worse it gets.

Also, it is barely noticeable when using PAS only. 

I will sent you a video later showing the fluctuations...

**Comment by papagino007** [comment]
Ok, I just did some test with bike on wheel stand.
I re-did a throttle calibration, doing is very slow to confirm very linear throttle response.
Then with bike on wheel stand, I did the following:

- Set FW to max 40
- Set mode to Torque
- Twisting throttle gently, I noticed that the motor reach basically maximum RPM at only 1/5th throttle position. Increasing throttle back and forth after this position doesn't change RPM (Remains to Max)
- The RPM fluctuation is very noticeable between 0 to 1/5th throttle.
- Setting mode to Speed render throttle usage almost unusable as it will cut off motor completely when speed related to throttle position is reached.

I will post a video shortly showing this...

**Comment by papagino007** [comment]
Video is currently being upload to YT as "Unlisted"...
Should be available shortly.

Here is the link:  https://youtu.be/YgbexzQHGxE

Also, something I have to mention is that my NXT display is not being supplied by the ASI BAC855 controller 5 volts output (Pin 11) , it is being supplied by a external 5V DC to DC converter. Doing this eliminates the 5V voltage sag from the controller (Pin11) as it is only rated at 50mA. You will notice that when I am doing the Throttle Calibration, my voltages is something like between 0.92V to 4.17V.

Hope this will help...

**Comment by electricrt** [comment]
Yes, this does make sense, and I have seen this kind of rpm fluctuation before. I'm gonna suggest you try to restore settings, once I upload the new beta firmware. I also suspect it has something to do with the PID settings.
